pkgs/applications/graphics/displaycal/default.nix

```
@@ -0,0 +1,74 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, xorg
+, certifi
+, dbus-python
+, build
+, wxPython_4_1
+, send2trash
+, PyChromecast
+, distro
+, makeWrapper
+, argyllcms
+, gtk3
+, gsettings-desktop-schemas
+}:
+
+buildPythonApplication rec {
+  pname = "displaycal";
+  version = "3.9.10";
+
+  src = fetchPypi {
+    pname = "DisplayCAL";
+    inherit version;
+    sha256 = "sha256-oDHDVb0zuAC49yPfmNe7xuFKaA1BRZGr75XwsLqugHs=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  propagatedBuildInputs = [
+    certifi
+    dbus-python
+    build
+    wxPython_4_1
+    send2trash
+    PyChromecast
+    distro
+  ];
+
+  buildInputs = with xorg; [
+    libX11
+    libXxf86vm
+    libXext
+    libXinerama
+    libXrandr
+  ];
+
+  pythonImportsCheck = [ "DisplayCAL" ];
+
+  doCheck = false; # Tests try to access an X11 session.
+
+  # no idea why it looks there - symlink .json lang (everything)
+  postInstall = ''
+    for x in $out/share/DisplayCAL/*; do
+      ln -s $x $out/lib/python3.10/site-packages/DisplayCAL
+    done
+
+    for binary in "$out/bin/"*; do
+      wrapProgram "$binary" \
+        --prefix PATH : "${lib.makeBinPath [ argyllcms ]}" \
+        --prefix PYTHONPATH : "$PYTHONPATH" \
+        --prefix XDG_DATA_DIRS : "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}"
+    done
+  '';
+
+  meta = with lib; {
+    description = "DisplayCAL Modernization Project (Migrated to Python 3)";
```

Recursive `description` is recursive?

```suggestion
    description = "Display calibration tool *migrated to Python 3)";
```

---

That's a reasonable point. DisplayCAL is a long-established name in the space it exists, so the migration of it just referencing it by name make sense in the context, but I can certainly see how a more generic name is better suited for users unfamiliar with the project--especially since if you search by name DisplayCAL should already show up.

---

Aside: `ikhal` via `khal` can get you a TUI calendar compatible with `vdirsyncer` (with the caveat that it's Python so it's slow and breaks frequently to where multiple people have overlays to disable `doCheck` because CI passing isn't a requirement for their upstream it seems).

---

The old Python 2.x version used this description.

> Display Calibration and Characterization powered by Argyll CMS

Though I would suggest using sentence case over random capital letters: "Display calibration and characterization powered by Argyll CMS"